### ISSUE TITLE
4.03 version

### ISSUE THREAD

**Issue by Drup** [janestreet/ppx_core#2]
I tried to fix ppx_core for 4.03 quickly ([here](https://github.com/Drup/ppx_core)), but there are some parts that I don't understand. Do you have one already done/Could you make one ?

**Comment by diml** [comment]
yeah we have one but. I created a trunk branch and pushed the patch

**Comment by Drup** [comment]
Great, thanks!

**Comment by Drup** [comment]
@diml I guess you have a version of ppx_deriving to go with that ?

**Comment by diml** [comment]
Do you mean ppx\_type\_conv? We don't use ppx\_deriving internally

**Comment by fpottier** [comment]
Hello,
I tried to install core_bench for ocaml 4.03+beta2, but the compilation of ppx_core fails with:
# ...[truncated]
# Error: This expression has type Types.constructor_arguments
#        but an expression was expected of type Types.type_expr list
Is a fix in the works?
Thanks!

**Comment by trefis** [comment]
We're in the process of getting 4.03-compatible version of our packages into opam: https://github.com/ocaml/opam-repository/pull/6094

If you're in a hurry you can always `opam pin` the 4.03 branch of the required repositories.

**Comment by little-arhat** [comment]
Hi, thank you for this library!

I wonder, why there is `failwith` in `type_is_recursive` for `Pcstr_record` in 4.03 version https://github.com/janestreet/ppx_core/blob/4.03/src/common.ml#L79 ? Unfortunately, this case prevents me from using `types_are_recursive` in my code and forces me to copy code, or use some other workaround. 

Are there plans for it to be fixed? 

Cheers!

**Comment by little-arhat** [comment]
Possible fix for above comment proposed in #5

**Comment by diml** [comment]
@little-arhat, basically for the initial 4.03 release we did whatever was necessary to get sexplib, core and other libraries to build without looking too much into the details. This was done to unblock testing with 4.03.

Once we'll support 4.03 properly internally we'll take care of properly supporting all 4.03 constructions

**Comment by agarwal** [comment]
Has there been any progress on supporting 4.03? AFAICT inline records, which were added in 4.03, still don't work with ppx_core. I get: `Failure("Pcstr_record not supported")`.

**Comment by trefis** [comment]
There hasn't been, but I will look into getting it done in the coming weeks.

**Comment by little-arhat** [comment]
@agarwal there is `113.33.01+4.03` version on opam, and it seems to work with `4.03` and `4.04`.

**Comment by trefis** [comment]
Ah indeed, I thought @agarwal was refering to https://github.com/janestreet/ppx_core/blob/master/src/gen/gen_ast_builder.ml#L126 .
Which has been fixed by https://github.com/janestreet/ppx_core/commit/a4344c7a370c551396b6b9bdd3f2a8d15ebac8b5 (on our 4.03 branch) and there is proper support on master since https://github.com/janestreet/ppx_core/commit/a4344c7a370c551396b6b9bdd3f2a8d15ebac8b5 .

**Comment by agarwal** [comment]
I reported to the wrong repo. The issue is with ppx_sexp_conv, and is already reported as janestreet/ppx_sexp_conv#9. Sorry for the noise.
